code/clue.js

```diff
@@ -1,59 +1,238 @@
+// SUSPECTS, WEAPONS AND ROOMS IN ARRAYS:
 
 const suspects = [
   mrGreen,
-  mrsWhite
-  // ...  and the rest
+  colonelMustard,
+  mrsPeacock,
+  professorPlum,
+  missScarlet,
+  mrsWhite,
 ]
 
-const weapons = []
+const weapons = [
+  rope,
+  knife,
+  candlestick,
+  dumbbell,
+  poison,
+  axe,
+  bat,
+  trophy,
+  pistol,
+]
 
-const rooms = []
+const rooms = ['Dining Room',
+  'Conservatory',
+  'Kitchen',
+  'Study',
+  'Library',
+  'Billiard Room',
+  'Lounge',
+  'Ballroom',
+  'Hall',
+  'Spa',
+  'Living Room',
+  'Observatory',
+  'Theater',
+  'Guest House',
+  'Patio',]
 
 // THIS FUNCTION WILL RANDOMLY SELECT ONE ITEM FROM THE ARRAY THAT YOU PASS IN TO THE FUNCTION.
-// YOU DON'T NEED TO CHANGE THIS, JUST TRY TO UNDERSTAND IT. AND HOW TO USE IT.
 const randomSelector = array => {
   return array[Math.floor(Math.random() * array.length)]
 }
 
 // CREATE AN OBJECT THAT KEEPS THE MYSTERY.
-// With a killer, a weapon and a room.
-// The values will be set later.
+let mystery = {
+  killer: null,
+  weapon: null,
+  room: null,
+}
 
 // This function will be invoked when you click on the killer card.
 const pickKiller = () => {
-  // This will randomly select a killer from the suspects. And add that to the mystery object.
   mystery.killer = randomSelector(suspects)
-
-  // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
   document.getElementById('killerCard').style.background = mystery.killer.color
-  document.getElementById(
-    'killerName'
-  ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
+  document.getElementById('killerName').innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
+  document.getElementById('killerAge').innerHTML = `${mystery.killer.age} years young`
+  document.getElementById('killerOccupation').innerHTML = mystery.killer.occupation
+  document.getElementById('killerDescription').innerHTML = mystery.killer.description
+  document.getElementById('killerImage').src = mystery.killer.image
+}
+document.getElementById('killerCard').onclick = pickKiller
+
+// FUNCTIONS pickWeapon 
+const pickWeapon = () => {
+  mystery.weapon = randomSelector(weapons)
+  document.getElementById('weaponName').innerHTML = mystery.weapon.name
+  document.getElementById('weaponFact').innerHTML = `${mystery.weapon.fact}`
+  document.getElementById('weaponWeight').innerHTML = `${mystery.weapon.weight} lbs`
+
 }
+document.getElementById('weaponCard').onclick = pickWeapon
 
-// CREATE FUNCTIONS pickWeapon and pickRoom in a similar way.
+//favouriteWeapon
+const favouriteWeapon = () => {
+  mystery.favouriteWeapon = randomSelector(weapons)
+}
+const shuffleFavouriteWeapon = () => {
+  mystery.favouriteWeapon = randomSelector(weapons)
+  document.getElementById('favouriteWeapon').innerHTML
+}
+shuffleFavouriteWeapon()
```

Comment:
So cool with the favourite weapon function. I did not get that far so I can’t give you a fair review on this part. But when I pressed the reveal button on your deployed site it worked perfectly. Great job Kat! 🥳